contxt/cli/commands/iot.py

```diff
@@ -113,3 +115,193 @@ def data(
+            grouping = groupings_by_label.get(grouping_name)
+
+            if not grouping:
+                print(f"Creating new grouping: {grouping_name}")
+                grouping = clients.iot.create_grouping(
+                    facility_id=feed.facility_id,
+                    label=grouping_name,
+                    description=grouping_name,
+                    is_public=True,
+                    field_category_id=[],
+                )
+                print(grouping)
+
+            print(f"Setting {len(field_ids)} fields to grouping {grouping_name}")
+            clients.iot.set_fields_for_grouping(grouping.id, field_ids)
+
+        print("Successful!")
+
+
+@iot.command()
+@click.argument("feed_key", type=str)
+@click.pass_obj
+def create_worksheet(clients: Clients, feed_key) -> None:
+    """Create Field Worksheets"""
+    fields = clients.iot.get_unprovisioned_fields_for_feed_key(feed_key=feed_key)
+    filename = f"{feed_key}_unprovisioned_worksheet.csv"
+
+    with open(os.path.join(".", filename), "w") as f:
+
+        headers = [
+            "Field Descriptor",
+            "Label",
+            "Data Type",
+            "Units",
+            "Scalar",
+            "Cumulative Value?",
+            "equipment group",
+            "Facility Main?",
+        ]
+        writer = csv.DictWriter(f, fieldnames=headers)
+        writer.writeheader()
+
+        for f in fields:
+            row = {
+                col_name: "" if col_name != "Field Descriptor" else f.field_descriptor  # type: ignore
+                for col_name in headers
+            }
+            writer.writerow(row)
+
+    print(f"Wrote unprovisioned fields to {filename}")
+
+
+@iot.command()
+@click.argument("facility_id", type=str)
+@click.argument("label", type=str)
+@click.argument("description", type=str)
+@click.argument("is_public", type=bool)
+@click.argument("field_category_id", type=str)
+@click.pass_obj
+def create_grouping(clients: Clients, facility_id, label, description, is_public, field_category_id):
+    """Create a new IOT Grouping"""
+    grouping = clients.iot.create_grouping(
+        facility_id=facility_id,
+        label=label,
+        description=description,
+        is_public=is_public,
+        field_category_id=[field_category_id],
+    )
+    print(grouping)
+
+
+@iot.command()
+@click.argument("grouping_id", type=str)
+@click.argument("field_id", type=str)
+@click.pass_obj
+def add_field_to_grouping(clients: Clients, grouping_id, field_id):
+    """Add field to grouping"""
+    grouping_field = clients.iot.add_field_to_grouping(grouping_id=grouping_id, field_id=field_id)
+    print(grouping_field)
+
+
+@iot.command()
+@click.argument("field_id_list", type=List[str])
+@click.pass_obj
+def unprovision_fields(clients: Clients, field_id_list: List[str]):
+    """Unprovision fields"""
+    for field_id in field_id_list:
+        clients.iot.unprovision_field(field_id)
```

Comment:
i would just accept an input file here and get rid of `unprovision_all_fields_in_feed`.

why? because you could just get the list from `contxt iot unprov-fields <feed_key>`